crates/core/tedge_agent/src/file_transfer_server/actor.rs

```diff
@@ -141,31 +144,26 @@ mod tests {
     }
 
     #[tokio::test]
-    #[serial_test::serial]
-    async fn check_server_does_not_panic_when_port_is_in_use() -> Result<(), anyhow::Error> {
+    async fn check_server_does_not_panic_when_port_is_in_use() -> anyhow::Result<()> {
         let ttd = TempTedgeDir::new();
 
         let http_config = HttpConfig::default()
             .with_data_dir(ttd.utf8_path_buf().into())
             .with_port(3746);
         let config_clone = http_config.clone();
 
-        // Spawn HTTP file transfer server
-        // handle_one uses port 3000.
-        let builder_one = FileTransferServerBuilder::new(http_config);
-        let handle_one = tokio::spawn(async move { builder_one.build().run().await });
-
-        // handle_two will not be able to bind to the same port.
-        let builder_two = FileTransferServerBuilder::new(config_clone);
-        let handle_two = tokio::spawn(async move { builder_two.build().run().await });
-
-        // although the code inside handle_two throws an error it does not panic.
-        tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+        // Both servers will attempt to bind to port 3746.
+        let server_one = FileTransferServerBuilder::new(http_config).build().run();
+        // This server will not be able to bind to the same port.
+        let server_two = FileTransferServerBuilder::new(config_clone).build().run();
 
-        // to check for the error, we assert that handle_one is still running
-        // while handle_two is finished.
-        assert!(!handle_one.is_finished());
-        assert!(handle_two.is_finished());
+        tokio::select! {
+            // Ensure we bind server_one first by polling that future first
+            biased;
+            res = server_one => bail!("expected second server to finish first, but first finished with: {res:?}"),
```

Comment:
```suggestion
            res = server_one => bail!("expected second server startup to fail first with port binding error, but first finished with: {res:?}"),
```

Reply:
This error message went away after refactoring the test to use a dynamic port